src/tools/perf/perftest_daemon.c

```diff
@@ -96,6 +107,288 @@ ucp_perf_daemon_server_conn_handle_cb(ucp_conn_request_h conn_request,
     }
 }
 
+static void
+ucp_perf_daemon_set_am_recv_handler(ucp_perf_daemon_context_t *ctx,
+                                    ucp_perf_daemon_am_id_t am_id,
+                                    ucp_am_recv_callback_t cb)
+{
+    ucp_am_handler_param_t param = {};
+    ucs_status_t status;
+
+    param.field_mask = UCP_AM_HANDLER_PARAM_FIELD_ID    |
+                       UCP_AM_HANDLER_PARAM_FIELD_CB    |
+                       UCP_AM_HANDLER_PARAM_FIELD_FLAGS |
+                       UCP_AM_HANDLER_PARAM_FIELD_ARG;
+    param.id         = am_id;
+    param.cb         = cb;
+    param.flags      = UCP_AM_FLAG_WHOLE_MSG;
+    param.arg        = ctx;
+    status           = ucp_worker_set_am_recv_handler(ctx->worker, &param);
+
+    ucs_assertv_always(status == UCS_OK, "status=%s", ucs_status_string(status));
+}
+
+static void
+ucp_perf_daemon_send_am_eager_reply(ucp_ep_h ep, ucp_perf_daemon_am_id_t am_id)
+{
+    ucp_request_param_t param = {};
+    ucs_status_ptr_t sptr;
+
+    param.op_attr_mask = UCP_OP_ATTR_FIELD_FLAGS;
+    param.flags        = UCP_AM_SEND_FLAG_REPLY | UCP_AM_SEND_FLAG_EAGER;
+
+    sptr = ucp_am_send_nbx(ep, am_id, NULL, 0ul, NULL, 0ul, &param);
+    if (UCS_PTR_IS_PTR(sptr)) {
+        ucp_request_free(sptr);
+    } else if (UCS_PTR_IS_ERR(sptr)) {
+        ucs_fatal("failed to send am id %u: %s", am_id,
+                  ucs_status_string(UCS_PTR_STATUS(sptr)));
```

Comment:
can we have better err handling?

Reply:
As discussed above, maybe we can postpone error handling until we support connection re-establishment?